src/game-coordinator/services/player-behavior-handler.service.ts

```diff
@@ -0,0 +1,54 @@
+import { ConfigurationService } from '@/configuration/services/configuration.service';
+import { GameState } from '@/games/models/game-state';
+import { PlayerConnectionStatus } from '@/games/models/player-connection-status';
+import { GamesService } from '@/games/services/games.service';
+import { PlayerSubstitutionService } from '@/games/services/player-substitution.service';
+import { PlayersService } from '@/players/services/players.service';
+import { Injectable, Logger } from '@nestjs/common';
+import { Cron, CronExpression } from '@nestjs/schedule';
+
+@Injectable()
+export class PlayerBehaviorHandlerService {
+  private logger = new Logger(PlayerBehaviorHandlerService.name);
+
+  constructor(
+    private readonly gamesService: GamesService,
+    private readonly playerSubstitutionService: PlayerSubstitutionService,
+    private readonly playersService: PlayersService,
+    private readonly configurationService: ConfigurationService,
+  ) {}
+
+  @Cron(CronExpression.EVERY_30_SECONDS)
+  async verifyPlayersJoinedGameServer() {
+    const bot = await this.playersService.findBot();
+    const timeout = (await this.configurationService.getTimeToJoinGameServer())
+      .value;
+    const gamesLive = await this.gamesService.getRunningGames();
+    await Promise.all(
+      gamesLive
+        .filter((game) => game.state === GameState.launching)
+        .filter((game) => Boolean(game.lastConfiguredAt))
```

Comment:
I don't like this line- it's hard to read. I'm a grinch for this- but are you trying to filter games that have been configured?

Reply:
Yes, exactly - if the game has not been configured, `lastConfiguredAt` returns `undefined`.

Reply:
OK, that reads odd to me for some reason. I don't have an alternative